examples/timeseries/event_classification_for_payment_card_fraud_detection.py

```diff
@@ -0,0 +1,430 @@
+"""
+
+
+def extract_features_and_label(example):
+    features = {k: example[k] for k in input_feature_names}
+    labels = tf.cast(example["TX_FRAUD"], tf.int32)
+    return features, labels
+
+
+# Target ratio of fraudulent transactions in the training dataset.
+target_rate = 0.2
+
+# Number of examples in a mini-batch.
+batch_size = 32
+
+train_ds = (
+    non_batched_train_ds.shuffle(10000)
+    .rejection_resample(
+        class_func=lambda x: tf.cast(x["TX_FRAUD"], tf.int32),
+        target_dist=[1 - target_rate, target_rate],
+        initial_dist=[1 - fraudulent_rate, fraudulent_rate],
+    )
+    .map(lambda _, x: x)  # Remove the label copy added by "rejection_resample".
+    .batch(batch_size)
+    .map(extract_features_and_label)
+    .prefetch(tf.data.AUTOTUNE)
+)
+
+# The test and validation dataset does not need resampling or shuffling.
+valid_ds = (
+    non_batched_valid_ds.batch(batch_size)
+    .map(extract_features_and_label)
+    .prefetch(tf.data.AUTOTUNE)
+)
+test_ds = (
+    non_batched_test_ds.batch(batch_size)
+    .map(extract_features_and_label)
+    .prefetch(tf.data.AUTOTUNE)
+)
+
+"""
+We print the first four examples of the training dataset. There is a simple way to identify some of the errors that could have been made above.
+"""
+
+for features, labels in train_ds.take(1):
+    print("features")
+    for feature_name, feature_value in features.items():
+        print(f"\t{feature_name}: {feature_value[:4]}")
+    print(f"labels: {labels[:4]}")
+
+"""
+## Train the model
+
+The original dataset is transactional, but the processed data is tabular and only contains normalized numerical values. Therefore, we train a feed-forward neural network.
+"""
+
+inputs = [keras.Input(shape=(1,), name=name) for name in input_feature_names]
+x = keras.layers.concatenate(inputs)
+x = keras.layers.Dense(32, activation="sigmoid")(x)
+x = keras.layers.Dense(16, activation="sigmoid")(x)
+x = keras.layers.Dense(1, activation="sigmoid")(x)
+model = keras.Model(inputs=inputs, outputs=x)
+
+"""
+Our goal is to differentiate between the fraudulent and legitimate transactions, so we use a binary classification objective. Because the dataset is imbalanced, accuracy is not an informative metric. Instead, we evaluate the model using the [area under the curve](https://en.wikipedia.org/wiki/Receiver_operating_characteristic#Area_under_the_curve) (AUC).
+"""
+
+model.compile(
+    optimizer=keras.optimizers.Adam(0.01),
+    loss=keras.losses.BinaryCrossentropy(),
+    metrics=[keras.metrics.Accuracy(), keras.metrics.AUC()],
+)
+model.fit(train_ds, validation_data=valid_ds)
+
+"""
+We evaluate the model on the test dataset."""
+
+model.evaluate(test_ds)
+
+"""With and AUC of ~83%, our simple fraud detector is showing encouraging results.
```

Comment:
How about showing how to save this model and talk a bit about the fact it's a pytorch / Jax / TF model

Reply:
I don't want to divert the core message too much. Examples on the keras website seem to be focused on demonstrating one thing (e.g., train a model on events in this case). Serialization and other things that can be done with keras have their own page (https://keras.io/api/models/model_saving_apis/). We will see if Keras' team wants more of this.

Reply:
You should be able to just do `model.save()`/ `load_model()` as usual.

Reply:
*Add section about model saving and model loading.